tensorboard/webapp/plugins/npmi/store/npmi_reducers.ts

```diff
@@ -80,6 +104,261 @@ const reducer = createReducer(
         },
       };
     }
+  ),
+  on(
+    actions.addSelectedAnnotations,
+    (state: NpmiState, {annotations}): NpmiState => {
+      let addedAnnotations: string[] = annotations.filter(
+        (annotation) => !state.selectedAnnotations.includes(annotation)
+      );
+      return {
+        ...state,
+        selectedAnnotations: [
+          ...state.selectedAnnotations,
+          ...addedAnnotations,
+        ],
+      };
+    }
+  ),
+  on(
+    actions.removeSelectedAnnotation,
+    (state: NpmiState, {annotation}): NpmiState => {
+      let annotationIndex = state.selectedAnnotations.indexOf(annotation);
+      return {
+        ...state,
+        selectedAnnotations: [
+          ...state.selectedAnnotations.slice(0, annotationIndex),
+          ...state.selectedAnnotations.slice(annotationIndex + 1),
+        ],
+      };
+    }
+  ),
+  on(
+    actions.setSelectedAnnotations,
+    (state: NpmiState, {annotations}): NpmiState => {
+      return {
+        ...state,
+        selectedAnnotations: annotations,
+      };
+    }
+  ),
+  on(
+    actions.clearSelectedAnnotations,
+    (state: NpmiState): NpmiState => {
+      return {
+        ...state,
+        selectedAnnotations: [],
+      };
+    }
+  ),
+  on(
+    actions.flagAnnotations,
+    (state: NpmiState, {annotations}): NpmiState => {
+      let notFlagged: string[] = annotations.filter(
```

Comment:
Another suggestion: would it be possible to change this to a positive word? I had to think harder (I am definitely spoiled by dumb code that I write).

Very dumb code I would write goes like:
```ts

const existingFlaggedAnnotations = new Set(state.flaggedAnnotations);

const newFlags: string[] = annotations.filter(
   (annotation) => existingFlaggedAnnotations.has(annotation)
);

if (newFlags.length) {
  return {
    ...state,
    flaggedAnnotations: [...existingFlaggedAnnotations, ...newFlags],
  };
}

// If all annotations are already flagged, user wants to remove them
for (const annotation of annotations) {
  existingFlaggedAnnotations.delete(annotation);
}

return {
 ...state,
 flaggedAnnotations: [...existingFlaggedAnnotations],
};
```

Reply:
Thank you for that tip.